### ISSUE TITLE
Still Maintained?

### ISSUE THREAD

**Issue by demisx** [ngbp/ngbp#230]
Hi, we are currently evaluating different AngularJS boilerplate generators and I can't really tell whether this project is still being maintained or not, since there hasn't been much commit activity lately. 

Is it safe to use this repo on a new project or would you recommend some other repo?

Thx.

**Comment by dzonatan** [comment]
I want to know this too.

**Comment by PaulL1** [comment]
The repository is still maintained, however it’s going through a reasonably sizeable upgrade.  The intent is to make the solution more pluggable and upgradeable, whilst potentially also moving to a gulp-based workflow instead of grunt-based, which would improve speed and maintainability.  That’s a significant rewrite, so things look a bit quiet, but it is still going on behind the scenes.

**Comment by msalahz** [comment]
Hi,
I have a question: Why you go with gulp-based instead of grunt-based workflow ?
Thanks,
Mohamed S.Zaghloul

**Comment by PaulL1** [comment]
That’s still a work in progress, but the feel is that gulp is the new thing that people are moving to.  The key benefits appear to be:
- easier to configure and maintain
- stream-based processing makes it more scalable and faster

**Comment by msalahz** [comment]
Thanks for the information, 

After checking glup docs I totally agree with you because I see glup much easier than grunt and this is a very good news for me as I'm planning to make my own workflow but that doesn't mean that i'm not looking forward to your upgrade :)

Thanks again and Good luck.
Mohamed S.Zaghloul

**Comment by demisx** [comment]
Can you please recommend any particular branch to use in the meantime (we are about 4 months from production)? Should I grab master?

**Comment by joshdmiller** [comment]
Hello, all! Yes, work is continuing on other branches, as @PaulL1 described.

As a minor clarification, the new version of ngbp will **not** be Gulp-based. It will use node streams as Gulp does and it will be able to make use of Gulp plugins, but it won't require or make use of Gulp directly. Gulp is a _very_ thin wrapper around some existing libraries that ngbp 0.4 will wrap a lot more heavily. It will be pretty sweet once we get it done.

However, I've been too busy this year to do much with it. If anyone is a node streams expert out there and wants to help out, ping me ([@joshdmiller](http://twitter.com/joshdmiller) or email) - it'd be a huge help. :-)

As for what branch to use, you'll need to use the 0.3.x branches as they're the only stable (read "functional") ones. I hope to push an update to the 0.3.x branch with some of the open pull requests within the next week, including porting to Angular 1.2 and fixing some bugs. If anyone wants to fix some open issues and submit pull requests, that would we be awesome, too.

I'll go ahead and close this issue, but feel free to keep the discussion going.

**Comment by dmackerman** [comment]
I was going to ask this myself. Honestly, gulp is great, but the current solution presented in this projects is still top notch as far as I'm concerned. The Gruntfile is large and quite daunting, but if you spend a day getting your head around how it works and does things the way it does - it's not that bad.

End of the day, these projects are meant as boilerplates to make you more productive. Developers love to spend time making sure that they choose the perfect tool for the job and I get that, but we've used this boilerplate with great success across many large (and small) apps within our company. 

This is one of the few boilerplates that I've found - and i've tried a ton of them - that follows the Angular recommended file structure for modules and then gets out of your way.
